channeldbservice/init.go

```diff
@@ -35,11 +36,47 @@ func Get(workingDir string) (db *channeldb.DB, cleanupFn func() error, err error
 		},
 	)
 	if err != nil {
-		return nil, nil, err
+		logger.Errorf("error getting service finished with error: %v", err)
+		dbName := "channel.db"
+		channeldb, err := bolt.Open(path.Join(workingDir, dbName), 0600, nil)
```

Comment:
I had defered the closing to the `removeKeyIfExists` function, but have moved it now. Can this interfere when we attempt to call the `Get` function if the db is not closed?

Reply:
I think I remember that LND itself needs an exclusive lock on this file, so that would mean there can be no other handle opening the same file.

Reply:
Ah, so that defer close might fail the call to Get on line 55 too https://github.com/breez/breez/pull/211/files#diff-8f00bd4dc5a013f42b0128a2a431c6997ae2b79aa4939862b60e267bb69da064R55